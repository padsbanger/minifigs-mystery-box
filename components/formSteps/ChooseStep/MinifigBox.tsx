import Image from "next/image";
import Link from "next/link";
import clsx from "clsx";

import { Minifig } from "@/lib/utils/services/MinifigsService/MinifigsService.types";
import Heading from "@/components/ui/Heading/Heading";

type MiniFigProps = Minifig & {
  isActive: boolean;
  onMinifigSelect: () => void;
};

const MinifigBox = ({
  set_img_url,
  name,
  set_url,
  isActive,
  onMinifigSelect,
}: MiniFigProps) => {
  return (
    <div
      onClick={onMinifigSelect}
      className={clsx(
        "hover:bg-red bg-white rounded-lg shadow-md cursor-pointer hover:p-2 transition-all my-2 md:m-0",
        {
          "shadow-selectedMinifig": isActive,
        },
      )}
    >
      <div className="flex align-baseline justify-center object-contain">
        <Image
          src={set_img_url}
          alt={name}
          width="320"
          height="320"
          placeholder="blur"
          className="w-full aspect-h-4 aspect-w-3 rounded-lg pt-4"
          blurDataURL="/placeholder.jpg"
        />
      </div>
      <Heading
        text={name}
        variant={"heading8"}
        level={6}
        classes="text-center !text-black normal-case m-4 font-bold line-clamp-1"
      />
      <Link
        href={set_url}
        target="_blank"
        className="text-center block m-4 font-plus-jakarta-sans font-bold text-blue-500 hover:text-blue-700"
      >
        Show Details
      </Link>
    </div>
  );
};

export default MinifigBox;
