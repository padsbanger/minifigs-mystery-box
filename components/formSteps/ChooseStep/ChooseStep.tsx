import Heading from "@/components/ui/Heading/Heading";
import { Minifig } from "@/lib/utils/services/MinifigsService/MinifigsService.types";
import {
  useMinifigsStore,
  useRandomMinifigs,
  useSelectedMinifig,
} from "@/lib/utils/services/MinifigsService/useMinifigsStore/useMinifigsStore";
import MinifigBox from "./MinifigBox";
import Button from "@/components/ui/Button/Button";

const ChooseStep = () => {
  const minifigs = useRandomMinifigs();
  const selectedMinifig = useSelectedMinifig();
  const setCurrentStep = useMinifigsStore.getState().setCurrentStep;
  const selectMinifig = useMinifigsStore.getState().onMinifigSelect;

  return (
    <div className="flex flex-col h-screen justify-center">
      <Heading
        text="Choose your minifig"
        variant={"heading3"}
        level={3}
        classes="text-center text-white"
      />
      <div className="container mx-auto">
        <div className="grid grid-cols-3 gap-6 my-32">
          {minifigs.map((el: Minifig) => (
            <MinifigBox
              {...el}
              key={el.name}
              isActive={selectedMinifig?.name === el.name}
              onMinifigSelect={() => selectMinifig(el)}
            />
          ))}
        </div>
        <div className="flex justify-center">
          <Button
            label={"Proceed to shipment!"}
            disabled={!selectedMinifig}
            onClick={() => setCurrentStep("SHIPPING")}
          />
        </div>
      </div>
    </div>
  );
};

export default ChooseStep;
