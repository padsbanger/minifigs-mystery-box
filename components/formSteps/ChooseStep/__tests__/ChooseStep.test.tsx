import { render } from "@testing-library/react";
import ChooseStep from "../ChooseStep";

const setup = () => render(<ChooseStep />);

describe("ChooseStep View", () => {
  test("should match snapshot", () => {
    const { asFragment } = setup();

    expect(asFragment()).toMatchSnapshot();
  });
});
