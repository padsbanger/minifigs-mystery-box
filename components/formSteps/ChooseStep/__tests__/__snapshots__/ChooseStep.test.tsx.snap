// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseStep View should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col h-auto md:h-screen justify-center"
  >
    <h3
      class="font-plus-jakarta-sans leading-normal uppercase text-headline-3 font-700 leading-[130%] text-center text-white"
    >
      Choose your minifig
    </h3>
    <div
      class="container mx-auto"
    >
      <div
        class="grid grid-cols-1 md:grid-cols-3 gap-6 my-32"
      />
      <div
        class="flex justify-center"
      >
        <button
          class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-6 px-12 rounded-md disabled:opacity-55 shadow-sm hover:shadow-xl transition-all active:shadow-2xl"
          disabled=""
        >
          <span
            class="uppercase font-plus-jakarta-sans"
          >
            Proceed to shipment!
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
